python/tank/deploy/tank_commands/unregister_folders.py

```diff
@@ -0,0 +1,311 @@
+        :param log: Logger
+        :param prompt: If true, the command may prompt the user for confirmation
+        :returns: List of dictionaries to represents the items that were unregistered.
+                  Each dictionary has keys path and sg_id.
+                  Note that the shotgun ids returned will refer to retired objects in 
+                  Shotgun rather than live ones.
+        """
+        log.debug("Unregister folders for Shotgun Entity %s..." % entity)
+    
+        # first of all, make sure we are up to date.
+        pc = path_cache.PathCache(self.tk)
+        try:
+            pc.synchronize(log)
+        finally:
+            pc.close()
+
+        # get the filesystem location ids which are associated with the entity    
+        sg_data = self.tk.shotgun.find(path_cache.SHOTGUN_ENTITY, [[path_cache.SG_ENTITY_FIELD, "is", entity]])
+        sg_ids = [x["id"] for x in sg_data]
+        log.debug("The following path cache ids are linked to the entity: %s" % sg_ids)
+        
+        if len(sg_ids) == 0:
+            log.info("This entity does not have any folder associated!")
+            return []
+        
+        return self._unregister_filesystem_location_ids(sg_ids, log, prompt)                              
+        
+        
+    def _unregister_filesystem_location_ids(self, ids, log, prompt):
+        """
+        Performs the unregistration of a path from the path cache database.
+        Will recursively unregister any child items parented to the given
+        filesystem location id.
+        
+        :param ids: List of filesystem location ids to unregister
+        :param prompt: Should the user be presented with confirmation prompts?
+        :param log: Logging instance
+        :returns: List of dictionaries to represents the items that were unregistered.
+                  Each dictionary has keys path and sg_id.
+                  Note that the shotgun ids returned will refer to retired objects in 
+                  Shotgun rather than live ones.
+        """
+        # now use the path cache to get a list of all folders (recursively) that are
+        # linked up to the folders registered for this entity.
+        
+        paths = []
+        pc = path_cache.PathCache(self.tk)
+        try:
+            for sg_fs_id in ids:
+                paths.extend(pc.get_folder_tree_from_sg_id(sg_fs_id))
+        finally:
+            pc.close()
+                
+        log.info("")
+        log.info("The following folders will be unregistered:")
+        for p in paths:
+            log.info(" - %s" % p["path"])
+        
+        log.info("")
+        log.info("Proceeding would unregister the above paths from Toolkit's folder database. "
+                 "This will not alter any of the content in the file system, but once you have "
+                 "unregistered the paths, they will not be recognized by Shotgun.")
+        log.info("")
```

Comment:
grammar tweaks and consistency:

> Proceeding will unregister the above paths from Toolkit's path cache. This will not alter any of the content on the file system ...

This is confusing...

>  ... they will not be recognized by Shotgun.

From a user perspective, what does that mean? Can we clarify this a bit? Specifically, it's Toolkit that won't recognize the paths... but then again, what will that mean? This is a tough one. We want to be clear but not over explain.

I'm also a bit bothered by the alternating use of "folder" and "path". It's the *path* cache but we're unregistering *folders*. That ship has sailed, but we should try and remain consistent where we can. :two: :moneybag: